app/components/navs/nav-rail.jsx

```
@@ -1,81 +1,112 @@
         return () => {
             window.removeEventListener('resize', handleResize);
         };
-    }, []);
+    }, [handleResize]);
 
     return (
-        <div className={`${props.className} nav-rail relative flex-col bg-[--md-sys-color-surface-container-low]  rounded-2xl nav-rail-${extended ? 'extended' : 'collapsed'}`}>
+        <div className={`${props.className} nav-rail ${screenWidth < 590 ? "hidden" : "flex"} w-full relative flex-col bg-[--md-sys-color-surface-container-low]  rounded-2xl nav-rail-${extended ? 'extended' : 'collapsed'}`}>
             <div className={`nav-rail-inner h-full nav-rail--padding-${extended ? 'extended' : 'collapsed'}`}>
-                <button className={`button selected-false rounded-full w-full`} role="button" onClick={() => { 
+                <button className={`extend-button selected-false rounded-full w-full`} role="button" onClick={() => { 
```

Biome (1.9.4)

[error] The elements with the following roles can be changed to the following elements:
<button>
<input type="button">


For examples and more information, see WAI-ARIA Roles

(lint/a11y/useSemanticElements)